### PR TITLE
feat!: loop

### PR DESCRIPTION
**Pull request by aljazerzen**
Work in progress.

Ref #407

### CODE REVIEW

**Comment by snth** [comment]
Oh this is exciting! Can't wait to try this out when it lands. Subscribed!

File: prql-compiler/src/test.rs [hunk 1]
```diff
@@ -3182,3 +3182,46 @@ a,b,c
...
+
+
+#[test]
+fn test_loop() {
+    assert_display_snapshot!(compile(r#"
+    from_text format:json '[{"n": 1 }]'
+    loop (
+        select n = n+1
+        filter n<5
+    )
```
Comment (max-sixty) [inline prql-compiler/src/test.rs:3237]: This does look quite nice!

File: book/src/internals/name-resolving.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ three things can happen:
 
 ## Translating to SQL
 
-When translating into a SQL statement which references only one table, there is
+When translating into an SQL statement which references only one table, there is
```
Comment (max-sixty) [inline book/src/internals/name-resolving.md:53]: Sorry to leave a comment criticizing the tiniest piece of this PR — but I think `a SQL` would be idiomatic. Or do others have thoughts?

Comment (max-sixty) [reply book/src/internals/name-resolving.md:53]: OK amazing, there's actually a whole debate out there! https://www.quora.com/Is-it-a-SQL-query-or-an-SQL-query

File: book/src/SUMMARY.md [hunk 3]
```diff
@@ -36,7 +36,9 @@
   - [Ranges](./language-features/ranges.md)
     <!--   - Regex — `REGEX_MATCH(foo, "\\w{3}")` -> `foo ~ r"\w{3}"`? Or -> `regex foo r"\w{3}"`? -->
   - [Regex](./language-features/regex.md)
-  - [Stdlib](./language-features/standard-library.md)
+  - [Standard library](./language-features/standard-library/README.md)
+    - [From text](./language-features/standard-library/from-text.md)
+    - [Loop](./language-features/standard-library/loop.md)
```
Comment (max-sixty) [inline book/src/SUMMARY.md:41]: Thanks, good idea

File: book/src/language-features/standard-library/loop.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Loop
+
+> _experimental_
```
Comment (max-sixty) [inline book/src/language-features/standard-library/loop.md:3]: +1

File: book/src/language-features/standard-library/loop.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+## Examples
+
+```prql
+from_text format:json '[{"n": 1 }]'
+loop (
+    filter n<4
+    select n = n+1
+)
+
+# returns [1, 2, 3, 4]
```
Comment (max-sixty) [inline book/src/language-features/standard-library/loop.md:35]: We could allow having actual results in the book, wouldn't be _that_ hard

File: book/src/internals/name-resolving.md (hunk 2 above)
Comment (aljazerzen) [reply book/src/internals/name-resolving.md:53]: Yeah, it depends on pronunciation. I'm saying S-Q-L, not Sequel, that's why this felt like a typo.

Comment (snth) [reply book/src/internals/name-resolving.md:53]: I would also say Sequel but I've heard Aljaz prefer to say S-Q-L during our presentation dry runs. 😄

Comment (max-sixty) [reply book/src/internals/name-resolving.md:53]: Saying "Sequel" is arguably more consistent with saying "Prequel"...!

**Pull request merged**